docs/sources/setup-grafana/configure-docker.md

```diff
@@ -57,48 +91,217 @@ The following settings are hard-coded when launching the Grafana Docker containe
+
+The `GRAFANA_VERSION` build argument must be a valid `grafana/grafana` Docker image tag. By default, Grafana builds an Alpine-based image. To build an Ubuntu-based image, append `-ubuntu` to the `GRAFANA_VERSION` build argument.
+
+Example:
+
+The following example shows you how to build and run a custom Grafana Docker image based on the latest official Ubuntu-based Grafana Docker image:
+
+```bash
+# go to the custom directory
+cd packaging/docker/custom
+
+# run the docker build command to build the image
+docker build
+  --build-arg "GRAFANA_VERSION=latest-ubuntu"
+  -t grafana-custom .
+
+# run the custom grafana container using docker run command
+docker run -d -p 3000:3000 --name=grafana grafana-custom
+```
+
+## Build a Grafana Docker image with pre-installed plugins
+
+To save time, you can customize a Grafana image by including plugins available on the [Grafana Plugin download page](/grafana/plugins). By doing so, you won't have to manually install the plugins each time, making the process more efficient.
+
+> **Note:** To specify the version of a plugin, you can use the `GF_INSTALL_PLUGINS` build argument and add the version number. The latest version is used if you don't specify a version number. For example, you can use `--build-arg "GF_INSTALL_PLUGINS=grafana-clock-panel 1.0.1,grafana-simple-json-datasource 1.3.5"` to specify the versions of two plugins.
+
+Example:
+
+The following example shows how to build and run a custom Grafana image with pre-installed plugins.
+
+```bash
+# go to the custom directory
+cd packaging/docker/custom
+
+# running the build command
+# include the plugins you want e.g. clock planel etc
+docker build
+  --build-arg "GRAFANA_VERSION=latest"
+  --build-arg "GF_INSTALL_PLUGINS=grafana-clock-panel,grafana-simple-json-datasource"
+  -t grafana-custom .
+
+# running the custom Grafana container using the docker run command
+docker run -d -p 3000:3000 --name=grafana grafana-custom
+```
+
+## Build a Grafana Docker image with pre-installed plugins from other sources
+
+You can create a Docker image containing a plugin that is exclusive to your organization, even if it is not accessible to the public. Simply use the `GF_INSTALL_PLUGINS` build argument to specify the plugin's URL and installation folder name, such as `GF_INSTALL_PLUGINS=;`.
+
+The following example demonstrates creating a customized Grafana Docker image that includes the clock panel and simple-json-datasource plugins. You can define these plugins in the build argument using the Grafana Plugin environment variable.
+
+```bash
+#go to the folder
+cd packaging/docker/custom
+
+#running the build command
+docker build
+--build-arg "GRAFANA_VERSION=latest"
+--build-arg "GF_INSTALL_PLUGINS=grafana-clock-panel, grafana-simple-json-datasource"
+-t grafana-custom .
+
+# running the docker run command
+docker run -d -p 3000:3000 --name=grafana grafana-custom
+```
+
+## Build Grafana with the Image Renderer plugin pre-installed
```

Comment:
This is likely worth moving up above the preinstalled plugins content, as it's something that does require using the custom image while plugins can be installed at boot time without needing a custom image.

Reply:
Done.